sklearn/feature_extraction/tests/test_text.py

```diff
@@ -859,6 +860,33 @@ def test_pickling_vectorizer():
             orig.fit_transform(JUNK_FOOD_DOCS).toarray())
 
 
+def test_countvectorizer_vocab_sets_when_pickling():
+    random.seed(0)
+    vocab_words = ['beer', 'burger', 'celeri', 'coke', 'pizza',
+                   'salad', 'sparkling', 'tomato', 'water']
+    for x in range(0, 100):
+        vocab_set = set(random.sample(vocab_words, 5))
+        cv = CountVectorizer(vocabulary=vocab_set)
+        unpickled_cv = pickle.loads(pickle.dumps(cv.fit(ALL_FOOD_DOCS)))
+        unpickled_cv.fit(ALL_FOOD_DOCS)
+        assert_equal(cv.get_feature_names(), unpickled_cv.get_feature_names())
+
+
+def test_countvectorizer_vocab_dicts_when_pickling():
+    random.seed(0)
```

Comment:
I used the `choice` method - if that's not what is conventionally used for scikit-learn, let me know.